File: second-year/second-semester/MPP/examen/rezolvari/examen_pb2/ExamenNetworking/src/main/java/exam/networking/rpcprotocol/Response.java
package exam.networking.rpcprotocol;

import java.io.Serializable;

public class Response implements Serializable {
    private ResponseType type;
    private Object data;

    private Response(){};

    public ResponseType getType(){
        return type;
    }

    public Object getData(){
        return data;
    }

    private void setType(ResponseType type){
        this.type=type;
    }

    private void setData(Object data){
        this.data=data;
    }

    @Override
    public String toString(){
        return "Response{" +
                "type='" + type + '\'' +
                ", data='" + data + '\'' +
                '}';
    }


    public static class Builder{
        private Response response=new Response();

        public Builder setType(ResponseType type) {
            response.setType(type);
            return this;
        }

        public Builder setData(Object data) {
            response.setData(data);
            return this;
        }

        public Response build() {
            return response;
        }
    }
}

File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/networking/Response.cs


namespace networking
{
    [Serializable]
    public class Response
    {
        public ResponseType type { get; set; }
        public Object data { get; set; }

        public Response()
        {
        }

        public override string ToString()
        {
            return "Response{" +
                   "type='" + type + '\'' +
                   ", data='" + data + '\'' +
                   '}';
        }
        public class Builder{
            private static Response response = new Response();

            public Builder Type(ResponseType type) {
                response.type = type;
                return this;
            }

            public Builder Data(Object data) {
                response.data = data;
                return this;
            }

            public Response Build() {
                return response;
            }
        }
    }    
}

